src/Stubs/StubFilesExtensionLoader.php

```diff
@@ -15,11 +15,21 @@ public function getFiles(): array
 		$typo3Version = new Typo3Version();
 		$versionParser = new VersionParser();
 
+		if ($versionParser->parseConstraints($typo3Version->getVersion())->matches($versionParser->parseConstraints('<= 11.3.0'))) {
+			$files[] = $stubsDir . '/ObjectStorage.stub';
```

Comment:
ObjectStorage annotations / implementation are still not correct in TYPO3 core we have to keep this (or parts of it)

Reply:
You are right. There are also parts of the Repository.stub that are not implemented in the core. I have a more general question however. How do we want to approach all those different TYPO3 versions with just one version of this package?

Let me give you an example:
`Repository::findAll()` is still wrong in the core. It says:

```php
     * @return QueryResultInterface|array
     * @phpstan-return QueryResultInterface|iterable<T>
```

That's wrong in a couple of ways because underlying
```php
return $this->createQuery()->execute();
```

returns `QueryResultInterface<int,T>`.

The right way to fix the core would be to this:

```php
     * @return QueryResultInterface
     * @phpstan-return QueryResultInterface<int, T>
```

So, we could add a stub for the current version only that fixes this but then the DynamicReturnTypeExtesion for `findAll()` fails because it doesn't expect the return type to have a key: https://github.com/sascha-egerer/phpstan-typo3/blob/master/src/Type/RepositoryFindAllDynamicReturnTypeExtension.php#L55 

It then falls back to resolving T based on the RepositoryClassName.

I'd like to contribute but it's really hard to do so for all the supported versions. Do you plan to cut off old braids sometime? I know everyone likes packages that work across several versions but I wouldn't know how to manage this.

Reply:
I'd really love the TYPO3 Core to get as many (and the same) annotations as provided by phpstan-typo3 (possibly even down to TYPO3 11LTS) so we can drop obsolete annotations/stubs from this package.

Reply:
I'm open for a discussion about how to support different TYPO3 Versions. So feel free to start a discussion with suggestions for possible solutions. But as @oliverklee already said the goal should be that the core solves most of this stuff bit itself so that there is no need to add stubs at all anymore.